imageio/core/functional_api.py

```diff
@@ -0,0 +1,98 @@
+from .imopen import imopen
+import numpy as np
+
+
+def imread(uri, *args, index=None, plugin=None, legacy_api=True,
+           extra_kwargs_plugin=None, **extra_kwargs_read):
+    """ Read an ndimage from a URI.
+
+    Opens the given URI and reads an ndimage from it. The exact behavior
+    depends on both, the file type and plugin used to open the file. To learn
+    about the exact behavior, check the documentation of the relevant plugin.
+    Typically, imread attempts to read all data stored in the URI.
+
+    Parameters
+    ----------
+    uri : {str, pathlib.Path, bytes, file}
+        The resource to load the image from, e.g. a filename, pathlib.Path,
+        http address or file object, see the docs for more info.
+    *args :
+        Additional positional arguments will be passed to the plugin instance
+        upon object creation.
+    index : {int, None}
+        If the URI contains multiple ndimages, select the index-th ndimage
+        from among them and return it. The exact behavior is plugin dependent.
+    plugin : {str, None}
+        The plugin to be used. If None, performs a search for a matching plugin.
+    legacy_api : {bool}
+        The API to be used. If true use the v2.9 API, otherwise use the new
+        v3.0 API.
```

Comment:
Can we be more clear about what the implications are?

Reply:
No longer applicable, as I've removed it. The idea was to only use the old plugins (`legacy_plugins=True`) or the new plugins (`True`). A better approach is to prefer the new plugins if possible and fall back to the legacy plugins otherwise. Takes less keyword arguments, too.